src/future/future.rs

```diff
@@ -105,6 +114,28 @@ extension_trait! {
     }
 
     pub trait FutureExt: std::future::Future {
+        /// Creates a future that is delayed before it starts yielding items.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// # async_std::task::block_on(async {
+        /// use async_std::future;
+        /// use std::time::Duration;
+        /// use async_std::future::FutureExt;
+        ///
+        /// let a = future::ready(1).delay(Duration::from_millis(2000));
+        /// dbg!(a.await);
+        /// # })
+        /// ```
+        #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+        #[cfg(any(feature = "unstable", feature = "docs"))]
+        fn delay(self, dur: Duration) -> DelayFuture<Self>
```

Comment:
We should obfuscate the type here, as we're awaiting `async fn` in traits to be released. Look at the `ret!` macro in the stream module

Reply:
If a concrete type is returned from a trait, people might start relying on it, making this a compatibility hazard towards a future `impl Future...` version.